pkg/util/kube.go

```diff
@@ -18,3 +30,62 @@ const (
 	// Kube annotation for podman volume image.
 	VolumeImageAnnotation = "volume.podman.io/image"
 )
+
+// ExtractPlayReader provide an io.Reader given a http.Request object
+// the function will extract the Content-Type header, if not provided, the body will be returned
+// of the header define a text format (json, yaml or text) it will also return the body
+// if the Content-Type is tar, we extract the content to the anchorDir and try to read the `play.yaml` file
+func ExtractPlayReader(anchorDir string, r *http.Request) (io.Reader, error) {
+	hdr, found := r.Header["Content-Type"]
+
+	// If Content-Type is not specific we use the body
+	if !found || len(hdr) == 0 {
+		return r.Body, nil
+	}
+
+	var reader io.Reader
+	switch hdr[0] {
+	// backward compatibility
+	case "text/plain":
+		fallthrough
+	case "application/json":
+		fallthrough
+	case "application/yaml":
+		fallthrough
+	case "application/text":
+		fallthrough
+	case "application/x-yaml":
+		reader = r.Body
+	case "application/x-tar":
+		// un-tar the content
+		err := archive.Untar(r.Body, anchorDir, nil)
+		if err != nil {
+			return nil, err
+		}
+
+		// check for play.yaml
+		path := filepath.Join(anchorDir, "play.yaml")
+		_, err = os.Stat(path)
+		// specify a message if the file does not exist
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, fmt.Errorf("file not found: tar missing play.yaml file at root")
+		} else if err != nil {
+			return nil, err
+		}
```

Comment:
just check the error from os.Open(), there is no need to stat+open

Reply:
Done